packages/profile-sync-controller/src/controllers/authentication/services.ts

```diff
@@ -0,0 +1,168 @@
+
+const ENV_URLS = getEnvUrls(Env.PRD);
+
+const AUTH_ENDPOINT: string = ENV_URLS.authApiUrl;
+export const AUTH_NONCE_ENDPOINT = `${AUTH_ENDPOINT}/api/v2/nonce`;
+export const AUTH_LOGIN_ENDPOINT = `${AUTH_ENDPOINT}/api/v2/srp/login`;
+
+const OIDC_ENDPOINT: string = ENV_URLS.oidcApiUrl || '';
+export const OIDC_TOKENS_ENDPOINT = `${OIDC_ENDPOINT}/oauth2/token`;
+const OIDC_CLIENT_ID = getOidcClientId(Env.PRD, Platform.EXTENSION);
+const OIDC_GRANT_TYPE = 'urn:ietf:params:oauth:grant-type:jwt-bearer';
+
+export type NonceResponse = {
+  nonce: string;
+};
+
+/**
+ * Auth Service - Get Nonce. Used for the initial JWTBearer flow
+ *
+ * @param publicKey - public key to associate a nonce with
+ * @returns the nonce or null if failed
+ */
+export async function getNonce(publicKey: string): Promise<string | null> {
+  const nonceUrl = new URL(AUTH_NONCE_ENDPOINT);
+  nonceUrl.searchParams.set('identifier', publicKey);
+
+  try {
+    const nonceResponse = await fetch(nonceUrl.toString());
+    if (!nonceResponse.ok) {
+      return null;
+    }
+
+    const nonceJson: NonceResponse = await nonceResponse.json();
+    return nonceJson?.nonce ?? null;
+  } catch (e) {
+    console.error('authentication-controller/services: unable to get nonce', e);
+    return null;
+  }
+}
+
+/**
+ * The Login API Server Response Shape
+ */
+export type LoginResponse = {
+  token: string;
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  expires_in: string;
+  /**
+   * Contains anonymous information about the logged in profile.
+   *
+   * @property identifier_id - a deterministic unique identifier on the method used to sign in
+   * @property profile_id - a unique id for a given profile
+   * @property metametrics_id - an anonymous server id
+   */
+  profile: {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    identifier_id: string;
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    profile_id: string;
+  };
+};
+
+/**
+ * Auth Service - Login. Will perform login with a given signature and will return a single use JWT Token.
+ *
+ * @param rawMessage - the original message before signing
+ * @param signature - the signed message
+ * @param clientMetaMetricsId - optional client metametrics id (to associate on backend)
+ * @returns The Login Response
+ */
+export async function login(
+  rawMessage: string,
+  signature: string,
+  clientMetaMetricsId: string,
+): Promise<LoginResponse | null> {
+  try {
+    const response = await fetch(AUTH_LOGIN_ENDPOINT, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        signature,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        raw_message: rawMessage,
+        metametrics: {
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          metametrics_id: clientMetaMetricsId,
+          agent: 'extension',
```

Comment:
shouldn't we receive the agent as param so we aren't tightly coupled to one single client/agent type?

Reply:
FFS, there are some other extension tie-ins as well that need replacing.
- Feature announcements - we don't want extension announcements
- Authentication - we want to say the agent is not just extension 

On it now.